api/v1alpha1/ciliumchaos_types.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +chaos-mesh:experiment
+
+// CiliumChaos is the control script's spec.
+type CiliumChaos struct {
```

Comment:
I notice that other chaos types have some test coverage here, e.g. https://github.com/form3tech-oss/chaos-mesh/blob/12bb0ed0fb6d1d89c181aeda0b131cc2f16fb544/api/v1alpha1/podchaos_types_test.go and  https://github.com/form3tech-oss/chaos-mesh/blob/12bb0ed0fb6d1d89c181aeda0b131cc2f16fb544/api/v1alpha1/podchaos_webhook_test.go

Is there some value in doing similar for CilliumChaos?  Especially with the selector / validation, perhaps we can add some coverage here?

Reply:
Definitely worth adding tests if we want to try to push this upstream :+1: